bgcval2/analysis_compare.py

```diff
@@ -4074,2135 +4138,169 @@ def CompareTwoRuns(jobIDA,
                 scatter=False)
 
 
+def load_comparison_yml(fn):
+    """
+    Load the config yaml.
+    TAkes an file path string 
+    Returns:
+        Details dict.
+    """
+    with open(fn, 'r') as openfile:
+        dictionary = yaml.safe_load(openfile)
+
+    details = {}    
+    details['name'] = dictionary.get('name', False)
+    details['jobs'] = dictionary.get('jobs', False) 
+
+    if not details['name']:
+        print('Please provide a name for your analysis. In your yaml, this is:')
+        print('name: MyAnalysisName')
+        exit(0)
+    if not details['jobs']:
+        print('Please provide at least one JobID for your analysis. In your yaml, this is:')
+        print('jobs: ')
+        print('    u-ab123:')
+        print('        description: "Job descrition"')
+        print('        colour: "red"')
+        print('        thickness: 0.7')
+        print("        linestyle: '-'")
+        print('        shifttime: 0.')
+        exit(0)       
+
+    details['do_analysis_timeseries'] = dictionary.get('do_analysis_timeseries', False) 
+    
+    details['master_suites'] = dictionary.get('master_suites', [])
+    
+    thicknesses = defaultdict(lambda: 0.75)
+    linestyles = defaultdict(lambda: '-')
```

Comment:
not a big fan of `defaultdict` - best way would be to check if the key exists, then if not, set it, and print a warning too: Setting missing key bla with bla

Reply:
Removed.